newIDE/app/src/EventsSheet/InstructionEditor/TextEditorDialog.js

```diff
@@ -0,0 +1,301 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import { t } from '@lingui/macro';
+
+import * as React from 'react';
+import Dialog from '../../UI/Dialog';
+import FlatButton from '../../UI/FlatButton';
+import HelpButton from '../../UI/HelpButton';
+import { Line, Column } from '../../UI/Grid';
+import ColorPicker from '../../UI/ColorField/ColorPicker';
+import MiniToolbar, { MiniToolbarText } from '../../UI/MiniToolbar';
+import SemiControlledTextField from '../../UI/SemiControlledTextField';
+
+const gd = global.gd;
+
+const toolbarItemStyle = {
+  marginRight: 10,
+};
+
+const styles = {
+  sizeTextField: {
+    width: 90,
+    ...toolbarItemStyle,
+  },
+  toolbarItem: toolbarItemStyle,
+  checkbox: toolbarItemStyle,
+};
+
+type Props = {|
+  event: any,
+  onClose: () => void,
+  onApply: () => void,
+|};
+
+type State = {|
+  textValue: any,
+  textColors: any,
+  backgroundColors: any,
+|};
+
+export default class TextEditorDialog extends React.PureComponent<
+  Props,
+  State
+> {
+  setName = () => {
+    const { event } = this.props;
+    const eventType = event.getType();
+    if (eventType === 'BuiltinCommonInstructions::Group') {
+      gd.asGroupEvent(event).setName(this.state.textValue);
+    }
+  };
+
+  setComment = () => {
+    const { event } = this.props;
+    const eventType = event.getType();
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      gd.asCommentEvent(event).setComment(this.state.textValue);
+    }
+  };
```

Comment:
Could you group these two methods in a single one called `_applyChangesOnEvent`?